es_manager.py

```diff
@@ -78,51 +78,53 @@ def bulk_query(self, index):
     arguments = parse_arguments()
 
     # Setup logging level
-    if arguments.debug:
-        logging.basicConfig(level=logging.DEBUG, format='%(message)s')
-    else:
-        logging.basicConfig(level=logging.INFO, format='%(message)s')
-
+    logging.basicConfig(
+        level=(logging.DEBUG if arguments.debug else logging.INFO)
+    )
+    logger = logging.getLogger(__name__)
+    # Set logging level to WARNING for the logger of elasticsearch package
     logging.getLogger('elasticsearch').setLevel(logging.WARNING)
 
     # create ES manager instance
     es_manager = ESManager(arguments.config)
 
     # Load data from build artifacts
-    locations = load_json('build/locations-combined.json')
-    services = load_json('build/services.json')
+    output_folder = 'build'
+    locations = load_json(f'{output_folder}/locations-combined.json')
+    services = load_json(f'{output_folder}/services.json')
 
     for index, docs in [('locations', locations), ('services', services)]:
-        new_ids = set()
-        old_ids = es_manager.old_ids[index]
+        current_ids = es_manager.current_ids[index]
+        artifact_ids = set()
+        create_ids = set()
+        update_ids = set()
+        delete_ids = set()
         for doc in docs:
             doc_id = doc['id']
-            new_ids.add(doc_id)
-            if doc_id not in old_ids:
-                # Perform a CREATE if document ID not in old ID set
-                logging.info(f'[CREATE] {index} {doc_id}')
+            artifact_ids.add(doc_id)
+            if doc_id not in current_ids:
+                # Perform a CREATE if document ID not in current ID set
+                logger.info(f'[CREATE] {index} {doc_id}')
+                create_ids.add(doc_id)
             else:
-                # Perform a UPDATE if document ID in old ID set
-                logging.info(f'[UPDATE] {index} {doc_id}')
+                # Perform a UPDATE if document ID in current ID set
+                logger.info(f'[UPDATE] {index} {doc_id}')
+                update_ids.add(doc_id)
             es_manager.create_or_update_doc(index, doc)
 
-        # Calculate ID status
-        create_ids = new_ids - old_ids
-        update_ids = new_ids.intersection(old_ids)
-        delete_ids = old_ids - new_ids
```

Comment:
Don't you still want this line? Otherwise, `delete_ids` will always be an empty set, right?

Reply:
right, not sure what I was doing, good catch.